File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/SearchHit.java
package com.example.grts;

import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.GraphQLUnion;

/**
 * Possible types: User, Order, Money
 */
public interface SearchHit extends GraphQLUnion {
    Type<SearchHit> Reflection = Type.ofClass(SearchHit.class);

    final class Fields implements TypeFields<SearchHit> {
        private Fields() {}

        public static final Field<SearchHit> __typename =
                Field.of("__typename", Reflection);
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/SearchHit.kt
@file:Suppress("warnings")

package viaduct.api.grts

import viaduct.apiannotations.InternalApi

@OptIn(InternalApi::class)
interface SearchHit : viaduct.api.types.Union {
  @OptIn(viaduct.apiannotations.InternalApi::class)
  object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.SearchHit> {
      override final val name = "SearchHit"
      override final val kcls = viaduct.api.grts.SearchHit::class
  }
  object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.SearchHit> {
          final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.SearchHit> =
              viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.SearchHit.Reflection)

  }

}